### ISSUE TITLE
Cmd_averagestats has UB

### ISSUE THREAD

**Issue by Bassoonian** [rh-hideout/pokeemerald-expansion#5000]
### Description

https://discord.com/channels/419213663107416084/774393519569502268/1263628135417774081

### Version

master (default, unreleased bugfixes)

**Comment by u8-Salem** [comment]
[luckytyphlosion pointed out](https://discord.com/channels/419213663107416084/774393519569502268/1263628657868406844) their solution would not be compatible with agbcc. I suggest we wait with this one until it is fully deprecated and removed.

**Comment by AlexOn1ine** [comment]
If there is a solution it wouldn't be bad to fix on 1.8

**Comment by u8-Salem** [comment]
> If there is a solution it wouldn't be bad to fix on 1.8

for which release is agbcc's deprecation sheduled? we should do this fix in the same release

**Comment by AlexOn1ine** [comment]
> > If there is a solution it wouldn't be bad to fix on 1.8
> 
> for which release is agbcc's deprecation sheduled? we should do this fix in the same release

it is being disabled in 1.9 but deprecated in 1.10. Though it would be a bugfix for 1.8.6 since this one is released at the same time 1.9 is

**Comment by mrgriffin** [comment]
A simple fix would be to write something like:
```c
inline u16 *GetBattlerStat(struct BattlePokemon *battler, u32 stat)
{
  switch (stat)
  {
  case STAT_ATTACK: return &battler->attack;
  case STAT_DEFENSE: return &battler->defense;
  ...
  default: return NULL;
  }
}

static void Cmd_averagestats(void)
{
    CMD_ARGS(u8 stat);

    u16 *stat1 = GetBattlerStat(&gBattleMons[gBattlerAttacker].attack), cmd->stat);
    u16 *stat2 = GetBattlerStat(&gBattleMons[gBattlerTarget].attack), cmd->stat);
    u16 avg = (*stat1 + *stat2) / 2;
    *stat1 = *stat2 = avg;
    *stat2 = avg;

    gBattlescriptCurrInstr = cmd->nextInstr;
}
```
